File: bridges/nvda/addon/globalPlugins/nvdaMcpBridge/domain/controllers/commands/ping.py
# nvdaMcpBridge domain -- PingHandler: liveness probe.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
# ROLE: command handler for `ping`, which reports whether speech is being withheld from the human.
# A ping must not reset the inactivity watchdog or the silence cap: a keepalive says nothing about testing.

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from .... import protocol
from .command_handler import CommandHandler

if TYPE_CHECKING:
	from .session_context import SessionContext


class PingHandler(CommandHandler):
	resets_inactivity = False

	def execute(self, ctx: SessionContext, request: protocol.Request) -> Any:
		suppressing = None
		if ctx.adapters is not None:
			suppressing = ctx.adapters.speech_source.is_suppressing()
		return protocol.PingResult(suppressing=suppressing)

File: bridges/voiceover/Sources/VoiceOverBridgeDomain/Controllers/Commands/Ping.swift
// ROLE: controller for `ping`, the liveness probe and the only handler that does not reset the
// command-inactivity watchdog.
//
// BUILT BY: Registry.
// `suppressing` is nil before the handshake, meaning "this bridge does not say", never false.

import ScreenReaderWire

public final class PingHandler: CommandHandler {
	public let resetsInactivity = false

	public init() {}

	public func execute(_ context: SessionContext, _ request: Request) throws -> any Encodable {
		PingResult(suppressing: context.adapters?.silenceControl.isSuppressing)
	}
}
